src/webots/nodes/WbMesh.cpp

```diff
@@ -0,0 +1,415 @@
+  aiNode *node = NULL;
+  unsigned int indexOffset = 0;
+  while (!queue.empty()) {
+    node = queue.front();
+    queue.pop_front();
+
+    // compute absolute transform of this node from all the parents
+    aiMatrix4x4 transform;
+    aiNode *current = node;
+    while (current != NULL) {
+      transform *= current->mTransformation;
+      current = current->mParent;
+    }
+
+    // merge all the meshes of this node
+    for (unsigned int i = 0; i < node->mNumMeshes; ++i) {
+      const aiMesh *mesh = scene->mMeshes[node->mMeshes[i]];
+
+      for (size_t j = 0; j < mesh->mNumVertices; ++j) {
+        // extract the coordinate
+        const aiVector3D vertice = transform * mesh->mVertices[j];
+        coordData[currentCoordIndex++] = vertice[0];
+        coordData[currentCoordIndex++] = vertice[1];
+        coordData[currentCoordIndex++] = vertice[2];
+        // extract the normal
+        const aiVector3D normal = transform * mesh->mNormals[j];
+        normalData[currentNormalIndex++] = normal[0];
+        normalData[currentNormalIndex++] = normal[1];
+        normalData[currentNormalIndex++] = normal[2];
+        // extract the texture coordinate
+        if (mesh->HasTextureCoords(0)) {
+          texCoordData[currentTexCoordIndex++] = mesh->mTextureCoords[0][j].x;
+          texCoordData[currentTexCoordIndex++] = mesh->mTextureCoords[0][j].y;
+        } else {
+          texCoordData[currentTexCoordIndex++] = 0.5;
+          texCoordData[currentTexCoordIndex++] = 0.5;
+        }
+      }
+
+      // create the index array
+      for (size_t j = 0; j < mesh->mNumFaces; ++j) {
+        const aiFace face = mesh->mFaces[j];
+        assert(face.mNumIndices == 3);
+        indexData[currentIndexIndex++] = face.mIndices[0] + indexOffset;
+        indexData[currentIndexIndex++] = face.mIndices[1] + indexOffset;
+        indexData[currentIndexIndex++] = face.mIndices[2] + indexOffset;
+      }
+
+      indexOffset += mesh->mNumVertices;
+    }
+
+    // add all the childrens of this node to the queue
+    for (size_t i = 0; i < node->mNumChildren; ++i)
+      queue.push_back(node->mChildren[i]);
+  }
+
+  if (!node) {
+    warn(tr("This file doesn't contain any mesh."));
+    return;
+  }
+
+  mTriangleMeshError = mTriangleMesh->init(coordData, normalData, texCoordData, indexData, totalVertices, 3 * totalFaces);
+
+  if (issueWarnings) {
+    foreach (QString warning, mTriangleMesh->warnings())
+      warn(warning);
+
+    if (!mTriangleMeshError.isEmpty())
+      warn(tr("Cannot create IndexedFaceSet because: \"%1\".").arg(mTriangleMeshError));
+  }
+}
+
+uint64_t WbMesh::computeHash() const {
+  const QByteArray meshPath = path().toUtf8();
+  return WbTriangleMeshCache::sipHash13x(meshPath.constData(), meshPath.size());
+}
+
+void WbMesh::exportNodeContents(WbVrmlWriter &writer) const {
+  if (!writer.isVrml()) {
+    WbTriangleMeshGeometry::exportNodeContents(writer);
+    return;
+  }
+  // Export the content as IndexedFaceSet in VRML
```

Comment:
```suggestion
  // export the content as IndexedFaceSet in VRML
```

just to be consistent with the other comments in this file